src/c_api.cpp

```diff
@@ -2053,35 +2113,11 @@ int LGBM_BoosterCalcNumPredict(BoosterHandle handle,
   API_END();
 }
 
-/*!
- * \brief Object to store resources meant for single-row Fast Predict methods.
- *
- * Meant to be used as a basic struct by the *Fast* predict methods only.
- * It stores the configuration resources for reuse during prediction.
- *
- * Even the row function is stored. We score the instance at the same memory
- * address all the time. One just replaces the feature values at that address
- * and scores again with the *Fast* methods.
- */
-struct FastConfig {
-  FastConfig(Booster *const booster_ptr,
-             const char *parameter,
-             const int predict_type_,
-             const int data_type_,
-             const int32_t num_cols) : booster(booster_ptr), predict_type(predict_type_), data_type(data_type_), ncol(num_cols) {
-    config.Set(Config::Str2Map(parameter));
-  }
 
-  Booster* const booster;
-  Config config;
-  const int predict_type;
-  const int data_type;
-  const int32_t ncol;
-};
 
 int LGBM_FastConfigFree(FastConfigHandle fastConfig) {
```

Comment:
should we rename this function?

Reply:
Is it not an issue to break backwards compatibility? I renamed internally because it makes more sense with regards to what it does now but opted to not rename public interfaces for that reason.

Reply:
we can add a new function,  and keep LGBM_FastConfigFree for now. And remove the old LGBM_FastConfigFree in the next version.

Reply:
So same thing for the two fast config init functions as well I suppose?

Reply:
Hello, sorry I didn't have time to review this carefully, but if we can keep the public API for now the better 👍 . Some external libraries were using this part of the API already.

Reply:
So I'm not sure I understand whether we should create the new functions *and* keep the old ones, or completely leave the public API as is.

Reply:
sorry for the late response. I think at least we should add a TODO tag here, and update this API name in next (or next next) official released version.

Reply:
6d949a0b910b1a6b633b228cc7947c2332d2bb95